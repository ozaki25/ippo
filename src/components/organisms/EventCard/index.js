import React from 'react';
import styled from 'styled-components';
import propTypes from 'prop-types';
import { Card, Elevation, Text } from '@blueprintjs/core';

const onClick = url => window.open(url, '_blank');

const StyledCard = styled(Card)`
  min-height: ${({ expand }) => (expand ? '130px' : 'inherit')};
`;

const EventCard = ({ title, eventUrl, catchMessage, place, datetime, interactive, expand }) => (
  <StyledCard
    expand={expand ? 1 : 0}
    interactive={interactive}
    onClick={() => onClick(eventUrl)}
    elevation={Elevation.TWO}
  >
    <h3 className="bp3-heading">{title}</h3>
    {catchMessage && <Text>{`概要：${catchMessage}`}</Text>}
    {place && <Text>{`場所：${place}`}</Text>}
    {datetime && <Text>{`日時：${datetime}`}</Text>}
  </StyledCard>
);

EventCard.displayName = 'EventCard';

EventCard.propTypes = {
  title: propTypes.string.isRequired,
  eventUrl: propTypes.string.isRequired,
  catchMessage: propTypes.string,
  place: propTypes.string,
  datetime: propTypes.string,
  interactive: propTypes.bool,
  expand: propTypes.bool,
};

EventCard.defaultProps = {
  catchMessage: '',
  place: '',
  datetime: '',
  interactive: true,
  expand: false,
};

export default EventCard;
