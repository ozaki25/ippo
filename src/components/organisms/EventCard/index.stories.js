import React from 'react';
import { text } from '@storybook/addon-knobs';
import EventCard from '.';

export default {
  title: 'organisms/EventCard',
};

export const タイトルとリンクのみの場合 = () => (
  <EventCard
    title={text('タイトル', 'イベントのタイトル')}
    eventUrl={text('url', 'https://connpass.com/')}
  />
);

export const 全項目ある場合 = () => (
  <EventCard
    title={text('タイトル', 'イベントのタイトル')}
    eventUrl={text('url', 'https://connpass.com/')}
    catchMessage={text('説明', 'イベントの説明です')}
    place={text('場所', '東京都千代田区丸の内')}
    datetime={text('日時', '2018年12月1日 19:30〜')}
  />
);

export const サイズを拡張する場合 = () => (
  <EventCard
    title={text('タイトル', 'イベントのタイトル')}
    eventUrl={text('url', 'https://connpass.com/')}
    expand
  />
);

export const 一行でおさめる場合 = () => (
  <EventCard
    title={text(
      'タイトル',
      'あいうえおかきくけこさしすせそたちつてとなにぬねのはひふへほまみむめもやゆよらりるれろわをん',
    )}
    eventUrl={text('url', 'https://connpass.com/')}
    catchMessage={text(
      '説明',
      'あいうえおかきくけこさしすせそたちつてとなにぬねのはひふへほまみむめもやゆよらりるれろわをん',
    )}
    place={text(
      '場所',
      'あいうえおかきくけこさしすせそたちつてとなにぬねのはひふへほまみむめもやゆよらりるれろわをん',
    )}
    datetime={text(
      '日時',
      'あいうえおかきくけこさしすせそたちつてとなにぬねのはひふへほまみむめもやゆよらりるれろわをん',
    )}
    noWrap
  />
);
