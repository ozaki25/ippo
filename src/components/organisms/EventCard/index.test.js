import React from 'react';
import { snapshot } from 'src/test/helpers';
import EventCard from '.';

const required = {
  title: 'イベントのタイトル',
  eventUrl: 'https://connpass.com/',
};

const optional = {
  title: 'イベントのタイトル',
  eventUrl: 'https://connpass.com/',
  catchMessage: 'イベントの説明です',
  place: '東京都千代田区丸の内',
  datetime: '2018年12月1日 19:30〜',
};

snapshot('EventCard/minimam', <EventCard {...required} />);

snapshot('EventCard/full', <EventCard {...required} {...optional} />);

snapshot('EventCard/expand', <EventCard {...required} expand />);
