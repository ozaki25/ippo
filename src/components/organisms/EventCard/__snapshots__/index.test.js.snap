// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot: EventCard/expand 1`] = `
.c0 {
  min-height: 130px;
}

<MockRouter
  createHref={[Function]}
  location=""
  params={Object {}}
  path=""
  push={[Function]}
  replace={[Function]}
  url=""
>
  <EventCard
    catchMessage=""
    datetime=""
    eventUrl="https://connpass.com/"
    expand={true}
    interactive={true}
    place=""
    title="イベントのタイトル"
  >
    <Styled(Blueprint3.Card)
      elevation={2}
      expand={1}
      interactive={true}
      onClick={[Function]}
    >
      <StyledComponent
        elevation={2}
        expand={1}
        forwardedComponent={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-bdVaJa",
              "isStatic": false,
              "lastClassName": "c0",
              "rules": Array [
                "
  min-height: ",
                [Function],
                ";
",
              ],
            },
            "displayName": "Styled(Blueprint3.Card)",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "sc-bdVaJa",
            "target": [Function],
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
            Symbol(Symbol.hasInstance): [Function],
          }
        }
        forwardedRef={null}
        interactive={true}
        onClick={[Function]}
      >
        <Blueprint3.Card
          className="c0"
          elevation={2}
          expand={1}
          interactive={true}
          onClick={[Function]}
        >
          <div
            className="bp3-card bp3-interactive bp3-elevation-2 c0"
            expand={1}
            onClick={[Function]}
          >
            <h3
              className="bp3-heading"
            >
              イベントのタイトル
            </h3>
          </div>
        </Blueprint3.Card>
      </StyledComponent>
    </Styled(Blueprint3.Card)>
  </EventCard>
</MockRouter>
`;

exports[`should match snapshot: EventCard/full 1`] = `
.c0 {
  min-height: inherit;
}

<MockRouter
  createHref={[Function]}
  location=""
  params={Object {}}
  path=""
  push={[Function]}
  replace={[Function]}
  url=""
>
  <EventCard
    catchMessage="イベントの説明です"
    datetime="2018年12月1日 19:30〜"
    eventUrl="https://connpass.com/"
    expand={false}
    interactive={true}
    place="東京都千代田区丸の内"
    title="イベントのタイトル"
  >
    <Styled(Blueprint3.Card)
      elevation={2}
      expand={0}
      interactive={true}
      onClick={[Function]}
    >
      <StyledComponent
        elevation={2}
        expand={0}
        forwardedComponent={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-bdVaJa",
              "isStatic": false,
              "lastClassName": "c0",
              "rules": Array [
                "
  min-height: ",
                [Function],
                ";
",
              ],
            },
            "displayName": "Styled(Blueprint3.Card)",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "sc-bdVaJa",
            "target": [Function],
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
            Symbol(Symbol.hasInstance): [Function],
          }
        }
        forwardedRef={null}
        interactive={true}
        onClick={[Function]}
      >
        <Blueprint3.Card
          className="c0"
          elevation={2}
          expand={0}
          interactive={true}
          onClick={[Function]}
        >
          <div
            className="bp3-card bp3-interactive bp3-elevation-2 c0"
            expand={0}
            onClick={[Function]}
          >
            <h3
              className="bp3-heading"
            >
              イベントのタイトル
            </h3>
            <Blueprint3.Text>
              <div
                className=""
              >
                概要：イベントの説明です
              </div>
            </Blueprint3.Text>
            <Blueprint3.Text>
              <div
                className=""
              >
                場所：東京都千代田区丸の内
              </div>
            </Blueprint3.Text>
            <Blueprint3.Text>
              <div
                className=""
              >
                日時：2018年12月1日 19:30〜
              </div>
            </Blueprint3.Text>
          </div>
        </Blueprint3.Card>
      </StyledComponent>
    </Styled(Blueprint3.Card)>
  </EventCard>
</MockRouter>
`;

exports[`should match snapshot: EventCard/minimam 1`] = `
.c0 {
  min-height: inherit;
}

<MockRouter
  createHref={[Function]}
  location=""
  params={Object {}}
  path=""
  push={[Function]}
  replace={[Function]}
  url=""
>
  <EventCard
    catchMessage=""
    datetime=""
    eventUrl="https://connpass.com/"
    expand={false}
    interactive={true}
    place=""
    title="イベントのタイトル"
  >
    <Styled(Blueprint3.Card)
      elevation={2}
      expand={0}
      interactive={true}
      onClick={[Function]}
    >
      <StyledComponent
        elevation={2}
        expand={0}
        forwardedComponent={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-bdVaJa",
              "isStatic": false,
              "lastClassName": "c0",
              "rules": Array [
                "
  min-height: ",
                [Function],
                ";
",
              ],
            },
            "displayName": "Styled(Blueprint3.Card)",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "sc-bdVaJa",
            "target": [Function],
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
            Symbol(Symbol.hasInstance): [Function],
          }
        }
        forwardedRef={null}
        interactive={true}
        onClick={[Function]}
      >
        <Blueprint3.Card
          className="c0"
          elevation={2}
          expand={0}
          interactive={true}
          onClick={[Function]}
        >
          <div
            className="bp3-card bp3-interactive bp3-elevation-2 c0"
            expand={0}
            onClick={[Function]}
          >
            <h3
              className="bp3-heading"
            >
              イベントのタイトル
            </h3>
          </div>
        </Blueprint3.Card>
      </StyledComponent>
    </Styled(Blueprint3.Card)>
  </EventCard>
</MockRouter>
`;
